sourcegraph/sourcegraph
# Improve banner and redirection logic for Sourcegraph App

Based on https://github.com/sourcegraph/sourcegraph/pull/49367

Prior to this PR in Sourcegraph App, if you run an instance with some local repositories, there was no notification about adding remote repositories (global nav bar notification) that would lead to the setup wizard. This PR adds this notification in case if you have no remote repositories 

## Test plan
- Run fresh Sourcegraph instance (not app instance)
- You should see the common flow be default (wizard is required unless you click to skip the wizard button)
- Run fresh Sourcegraph instance 
- If you run this and there are already some local repositories, you should see the home page but with a global setup banner with a link to the setup wizard
- After you add some remote repositories, you should see no banner 



## App preview:

- [Web](https://sg-web-vk-fix-app-setup-banner.onrender.com/search)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.
